src/client_side.cc

```diff
@@ -1638,7 +1638,10 @@ clientProcessRequest(ConnStateData *conn, const Http1::RequestParserPointer &hp,
 #endif
 
     if (internalCheck(request->url.path())) {
-        if (internalHostnameIs(request->url.host()) && request->url.port() == getMyPort()) {
+        const bool allowLocalhost = (conn->port->s.isLocalhost() || conn->port->s.isAnyAddr());
+        const bool isMyHostname = (allowLocalhost && strcmp(request->url.host(), "localhost") == 0) ||
+                                  internalHostnameIs(request->url.host());
+        if (isMyHostname && request->url.port() == conn->port->s.port()) {
```

Comment:
Please move all these checks inside one dedicated function.

FWIW, I recommend starting that method with the universal (and fast) port number check. It will simplify the function code structure because you can simply `return false` after that check fails and `return true` from any confirmed alternative (validated after that port number check). Adding debugging to those `return` statements may be a good idea (and will help self-document that code).